src/syrupy/matchers.py

```diff
@@ -30,29 +34,30 @@ def path_type(
     *,
     types: Tuple["PropertyValueType", ...] = (),
     strict: bool = True,
-    regex: bool = False,
+    replacer: "Replacer" = lambda data, _: Repr(AmberDataSerializer.object_type(data)),
+    **_: Any,
```

Comment:
Is this to make it easy to extend the API?

Reply:
Yup and maintain backwards compatibility with the removal of the `regex` kwarg.